src/samples/coap/lwm2m-bs-server.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * This file is part of the Soletta (TM) Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+   To run: ./lwm2m-sample-bs-server
+   For every LWM2M client that connects with the bootstrap server, the bootstrap
+   server will send bootstrap information in order for that client to connect
+   with the lwm2m-sample-server.
+ */
+
+#include "sol-lwm2m.h"
+#include "sol-mainloop.h"
+#include "sol-vector.h"
+#include "sol-util.h"
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <errno.h>
+
+#define LIFETIME (60)
+
+#define SERVER_OBJ_ID (1)
+#define SERVER_OBJ_SHORT_RES_ID (0)
+#define SERVER_OBJ_LIFETIME_RES_ID (1)
+#define SERVER_OBJ_BINDING_RES_ID (7)
+#define SERVER_OBJ_REGISTRATION_UPDATE_RES_ID (8)
+
+#define SECURITY_SERVER_OBJ_ID (0)
+#define SECURITY_SERVER_SERVER_URI_RES_ID (0)
+#define SECURITY_SERVER_IS_BOOTSTRAP_RES_ID (1)
+#define SECURITY_SERVER_SERVER_ID_RES_ID (10)
+#define SECURITY_SERVER_CLIENT_HOLD_OFF_TIME_RES_ID (11)
+#define SECURITY_SERVER_BOOTSTRAP_SERVER_ACCOUNT_TIMEOUT_RES_ID (12)
+
+const char *known_clients[] = { "cli1", "cli2", NULL };
+
+static struct sol_blob server_one_addr = {
+    .type = &SOL_BLOB_TYPE_NO_FREE,
+    .parent = NULL,
+    .mem = (void *)"coap://localhost:5683",
+    .size = sizeof("coap://localhost:5683") - 1,
+    .refcnt = 1
+};
+
+static struct sol_blob binding = {
+    .type = &SOL_BLOB_TYPE_NO_FREE,
+    .parent = NULL,
+    .mem = (void *)"U",
+    .size = sizeof("U") - 1,
+    .refcnt = 1
+};
+
+static void
+write_resource_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *bs_cinfo, const char *path,
+    enum sol_coap_response_code response_code)
+{
+    char *name = strdup(sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo));
+    int r;
+
+    if (response_code != SOL_COAP_RESPONSE_CODE_CHANGED) {
+        fprintf(stderr, "The client %s could not write the object(s)/resource at %s.\n",
+            name, path);
+        return;
+    }
+
+    printf("The client %s wrote the object(s)/resource at %s.\n", name, path);
+
+    r = sol_lwm2m_bootstrap_server_send_finish(server, bs_cinfo);
+    if (r < 0) {
+        fprintf(stderr, "Could not send Bootstrap Finish\n");
+    } else
+        printf("Client-initiated Bootstrap from %s finished!\n", name);
```

Comment:
Move this msg the above `sol_lwm2m_bootstrap_server_send_finish` and change the message to
`printf("Sending finish bootstrap finish to %s", name);`. Feel free to change the message content